schema.sql

```diff
@@ -0,0 +1,56 @@
+CREATE TABLE users (
+	-- users contains all user-data. An entry in this table corresponds to one
+	-- specific person. The account balance is reconstructed completely out of the
+	-- transactions table, to reduce duplication of information.
+
+
+	-- user_id is a sequential identifier.
+	user_id INTEGER NOT NULL,
+	-- name is the username used for display and login.
+	name TEXT,
+	-- password is a bcrypt-hashed password.
+	password BINARY,
+
+	-- constraints
+	PRIMARY KEY (user_id)
+);
+
+CREATE TABLE cards (
+	-- cards contains all card-data. An entry in this table corresponds to one
+	-- physical card. Every user can have an arbitrary number of cards.
+
+
+	-- card_id is a sequential identifier.
+	card_id BINARY NOT NULL,
```

Comment:
Hm, ein Vorteil von einer separaten column ist, dass man Platz spart in der transaction-table. Das könnte mich dann doch noch überzeugen, dass das ganz sinnig sein könnte.